.github/continuous-integration.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+if [[ -d "${PERSIST_DIR}/ccache-cachedir" ]]; then
+    echo "The Ccache directory exists at ${PERSIST_DIR}/ccache-cachedir"
+else
+    echo "Creating Ccache directory at ${PERSIST_DIR}/ccache-cachedir"
+    mkdir -p ${PERSIST_DIR}/ccache-cachedir
+    envsubst > ${PERSIST_DIR}/ccache-config <<EOF
+    max_size = 20G
+    cache_dir = ${PERSIST_DIR}/ccache-cachedir
+EOF
+fi
+
+(
+  cd arc-mlir;
+  BUILD_FLAVOUR=Release A2M_ASSERTS="1" ./arc-mlir-build;
+  ninja tools/arc-mlir/arc-check -C $A2M_BUILD/llvm-build/;
+)
+
+(
+  cd arc-script;
```

Comment:
We need a printout before each step. If things fail, that will greatly help locating the step which failed, especially if some steps execute silently.

If you add

```
function run-step {
    echo "Running $@"
    "$@"
}
```
at the top of the file, you can do:
```
run-step  cargo insta test --package=arc-script-test-compile
run-step  cargo insta accept;
...
```